### PR TITLE
Remove deepcopy on ResultSet _init_

### PR DESCRIPTION
**Pull request by andryw**
deepcopy on ResultSet slow down performance.

### CODE REVIEW

File: esengine/bases/result.py [hunk 1]
```diff
@@ -4,17 +4,18 @@
 import elasticsearch.helpers as eh
 from six import text_type
 
+HITS = 'hits'
```
Comment (ederfmartins) [inline esengine/bases/result.py]: Add a unit test to document this magic constant.

File: esengine/bases/result.py [hunk 2]
```diff
@@ -22,6 +23,13 @@ def __init__(self, resp, model, query=None,
     def __iter__(self):
         return self.values
 
+    def _extract_meta(self, resp):
```
Comment (ederfmartins) [inline esengine/bases/result.py]: Add a unit test to this method :+1:

File: esengine/bases/result.py (hunk 1 above)
Comment (rochacbruno) [inline esengine/bases/result.py]: Why this exists?

Comment (andryw) [inline esengine/bases/result.py]: The string 'hits' is used in many places. If this value change, it is need to change the value in every place. 
So it's easier to maintain and understand.

**Pull request merged**